ngraph/core/src/op/prior_box_clustered.cpp

```diff
@@ -96,13 +96,31 @@ shared_ptr<Node> op::PriorBoxClustered::clone_with_new_inputs(const OutputVector
 
 bool op::PriorBoxClustered::visit_attributes(AttributeVisitor& visitor)
 {
-    visitor.on_attribute("widths", m_attrs.widths);
-    visitor.on_attribute("heights", m_attrs.heights);
+    float step = 0;
+    float step_w_tmp = m_attrs.step_widths;
+    float step_h_tmp = m_attrs.step_heights;
+
+    visitor.on_attribute("step", step);
+    visitor.on_attribute("step_w", m_attrs.step_widths);
+    visitor.on_attribute("step_h", m_attrs.step_heights);
+    if (step != 0)
+    {
+        // deserialization:
+        // if step_w/h is 0 or did not change, replace it with step
+        if (m_attrs.step_widths == 0 || m_attrs.step_widths == step_w_tmp)
+        {
+            m_attrs.step_widths = step;
+        }
+        if (m_attrs.step_heights == 0 || m_attrs.step_heights == step_h_tmp)
+        {
+            m_attrs.step_heights = step;
+        }
```

Comment:
Why this logic is in the visit attributes?

Reply:
So this op can have three parameters that relate to each other: _step_, _step_w_, _step_h_.
The IR can contain 1(_step_) or 2(_step_w_, _step_h_) or 3(_step_, _step_w_, _step_h_) of them.

- If the IR has _step_w_ and _step_h_ specified we use that. 
- If _step_w_ or _step_h_ is 0 we replace it with the value of step.
- If _step_ = 0 and _step_w_==_step_h_ we assign _step_w's_ value to step.
- Only step is present, assign its value to _step_w_ and _step_h_.

This logic is fairly easy to do, just not in visit attributes, where we have to make a universal solution for all uses of this method (due to the nature of this pattern, we don't know if _visit_attributes_ is being run on serialization/deserialization, or conversion to CNNN path).

So without this logic, when we're reading the IR and we have only step parameter available in the IR, we would not initialize _step_w_ and _step_h_, which would give the wrong result as the op would have wrong parameters.

This could all be made much more clear if there was a way to distinguish the path we're on or overload _visit_attributes_ method for _XMLDeserializer_, but AFAIK it's not possible.